### ISSUE TITLE
[Bug] pre_hook executed two times

### ISSUE THREAD

**Issue by rokity** [dbt-athena/dbt-athena#579]
### Is this a new bug in dbt-athena?

- [X] I believe this is a new bug in dbt-athena
- [X] I have searched the existing issues, and I could not find an existing issue for this bug

### Current Behavior

Analyzing Athena queries we noticed that the pre_hook was executed two times instead that one, it sounds strange.
The current code.

```sql
 -- depends_on: {{ ref('seasons') }}
{{ config(
    materialized='incremental',
    table_type='iceberg',
    incremental_strategy='append',
    unique_key=["SEASON_YEAR_CODE"],
    external_location='*',
    write_compression='SNAPPY',
    partitioned_by=['SEASON_YEAR_CODE'],
    pre_hook = [
        """
        {% if is_incremental() %}
        delete from {{this}}
        where 
        SEASON_YEAR_CODE in ( select SEASON_YEAR_CODE from {{ ref('seasons') }} )
        {% endif %}
        """
    ]
) }}

select * from bronze 
```
Athena export csv :

```csv
delete from silver  where SEASON_YEAR_CODE in ( select SEASON_YEAR_CODE from seasons ),2024-02-07T11:11:05.531+01:00 
delete from silver  where SEASON_YEAR_CODE in ( select SEASON_YEAR_CODE from seasons ),2024-02-07T11:10:43.883+01:00
```

### Expected Behavior

The prehook must to be executed just one time, not two.

### Steps To Reproduce

1. Environment dbt-athena last version take from github repo
2. threads: 16 ,      num_retries: 6 on profile.yaml

The Data Modeling it's just a Delete-Insert strategy, we delete all the records with season_code present on ref=seasons and we append the one in bronze table.

### Environment

```markdown
- OS:MWAA Airflow
- Python: N/A
```

**Comment by svdimchenko** [comment]
Hi @rokity. Could you specify the adapter version ? Seems the bug is fixed in `v1.7.2` which was released yesterday. Here is the [PR](https://github.com/dbt-athena/dbt-athena/pull/575)

**Comment by rokity** [comment]
Hi @svdimchenko I just noticed now that on the last release, it was fixed, thanks for your work.
